cmd/poller/plugin/changelog/changelog.go

```diff
@@ -243,12 +267,53 @@ func (c *ChangeLog) Run(dataMap map[string]*matrix.Matrix) ([]*matrix.Matrix, *u
 	return matricesArray, nil, nil
 }
 
+// CompareMetrics compares the metrics of the current and previous instances
+func (c *ChangeLog) CompareMetrics(curMat *matrix.Matrix) map[string]map[string][2]float64 {
+	metricChanges := make(map[string]map[string][2]float64)
+	prevMat := c.previousData
+
+	met := maps2.Keys(c.previousData.GetMetrics())
+
+	for _, metricKey := range met {
+		prevMetric := prevMat.GetMetric(metricKey)
+		curMetric := curMat.GetMetric(metricKey)
+		for key, currInstance := range curMat.GetInstances() {
+			prevInstance := prevMat.GetInstance(key)
+			prevIndex := prevInstance.GetIndex()
+			currIndex := currInstance.GetIndex()
+			curVal := curMetric.GetValues()[currIndex]
+			prevVal := prevMetric.GetValues()[prevIndex]
+			if prevInstance != nil {
+				if curVal != prevVal {
+					if _, ok := metricChanges[key]; !ok {
+						metricChanges[key] = make(map[string][2]float64)
+					}
+					metricChanges[key][metricKey] = [2]float64{prevVal, curVal}
+				}
+			}
+		}
+	}
+	return metricChanges
+}
+
 // copyPreviousData creates a copy of the previous data for comparison
 func (c *ChangeLog) copyPreviousData(cur *matrix.Matrix) {
 	labels := c.changeLogConfig.PublishLabels
-	labels = append(labels, c.changeLogConfig.Track...)
+	var met []string
+	for _, t := range c.changeLogConfig.Track {
+		mKey := cur.DisplayMetricKey(t)
+		if mKey != "" {
```

Comment:
minor but it's better to prefer the positive equality instead of the negative so
```go
if mKey == "" {
    labels = append(labels, t)
		} else {
    met = append(met, mKey)		
		}
```